pkg/label/label.go

```diff
@@ -20,25 +20,25 @@ const (
 	Watching = "app-operator.giantswarm.io/watching"
 )
 
+// AppVersionSelector returns the label selector for this instance of
+// app-operator.
 func AppVersionSelector(unique bool) labels.Selector {
-	version := GetProjectVersion(unique)
-	s := fmt.Sprintf("%s=%s", label.AppOperatorVersion, version)
+	var selector string
 
-	selector, err := labels.Parse(s)
+	if unique {
+		// Unique instance watches all namespaces for app CRs with the unique
+		// app version (0.0.0).
+		selector = fmt.Sprintf("%s=%s", label.AppOperatorVersion, project.ManagementClusterAppVersion())
+	} else {
+		// Other instances watch the namespace they are running in but exclude
+		// unique app CRs.
+		selector = fmt.Sprintf("%s!=%s", label.AppOperatorVersion, project.ManagementClusterAppVersion())
```

Comment:
For workload clusters the namespace will include app-operator's own app CR. 

We exclude that using the selector because its a unique app CR.